### PR TITLE
Fix passing the local classes in partials

### PR DESCRIPTION
**Pull request by aleph1ow**
## Description

Passing the local `classes` to partials was not properly done.
As it currently is, the partials would have not received the local `classes` if it was used.

## Motivation and Context

This pr addresses and fix all the partials that can accept classes in input.

## How Has This Been Tested?

Every modified partial has been checked by passing the `classes` local to the render method that uses it and verifying that classes are outputted correctly.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: app/views/spree/components/typography/_hero_title.html.erb [hunk 1]
```diff
@@ -2,10 +2,10 @@
...
 
   # Classes
-  class_names = [base_class]
-  class_names = class_names.join(" ")
+  classes = local_assigns.fetch(:classes, [])
+  class_names = classes.push(base_class).join(" ")
 %>
 
-<h1 class="hero-title">
+<h1 class="<%= class_names%>">
```
Comment (kennyadsl) [inline app/views/spree/components/typography/_hero_title.html.erb:9]: Can you please add a space after `class_names` here?

File: app/views/spree/components/typography/_subtitle.html.erb [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
 
   # Classes
-  class_names = [base_class]
-  class_names = class_names.join(" ")
+  classes = local_assigns.fetch(:classes, [])
+  class_names = classes.push(base_class).join(" ")
 %>
 
-<p class="subtitle">
+<p class="<%= class_names%>">
```
Comment (kennyadsl) [inline app/views/spree/components/typography/_subtitle.html.erb:9]: Can you please add a space after `class_names` here?

**Comment by kennyadsl** [comment]
Restarted the CI some times but it keeps failing :(

File: app/views/spree/components/buttons/_button_primary.html.erb [hunk 3]
```diff
@@ -11,11 +11,12 @@
   type = local_assigns.fetch(:type, 'button')
 
   # Classes
-  class_names = [base_class]
-  class_names.push("#{base_class}--bordered") if bordered
-  class_names.push("#{base_class}--full-width") if full_width
-  class_names.push("#{base_class}--small") if small
-  class_names = class_names.join(" ")
+  classes = local_assigns.fetch(:classes, [])
```
Comment (blocknotes) [inline app/views/spree/components/buttons/_button_primary.html.erb:14]: I think that we could move this line above in the _Optional props_ list to preserve the logic (the same comment is valid for the whole PR).

File: app/views/spree/components/layout/_top_bar.html.erb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <div class="top-bar">
   <%= render partial: 'spree/components/brand/logo' %>
   <div class="top-bar__search">
-    <%= render partial: 'spree/components/search/search_bar' %>
+    <%= render 'spree/components/search/search_bar' %>
```
Comment (ChristianRimondi) [inline app/views/spree/components/layout/_top_bar.html.erb:4]: Why did you remove the `partial` keyword? 🤔

File: app/views/spree/components/search/_products_search.html.erb [hunk 5]
```diff
@@ -7,14 +7,14 @@
...
-  class_names = class_names.join(" ")
+  classes = local_assigns.fetch(:classes, [])
+  class_names = classes.push(base_class).join(" ")
 %>
 
 <%= form_tag spree.products_path, class: class_names, method: :get do %>
   <% cache [I18n.locale, @taxons] do %>
     <%= render(
-      "spree/components/forms/inputs/select",
+      "spree/components/forms/inputs/select_input",
```
Comment (ChristianRimondi) [inline app/views/spree/components/search/_products_search.html.erb:17]: I think that is better to move this change in a different commit. `Fix passing the local classes in partials` doesn't describe why this change is required.

File: app/views/spree/components/layout/_top_bar.html.erb (hunk 4 above)
Comment (aleph1ow) [reply app/views/spree/components/layout/_top_bar.html.erb:4]: I'm going to remove this change. I'll do this globally in another pr
